### DISCUSSION TITLE
ReinbowKit & Metamask Mobile not connecting, Using @walletconnect/universal-provider 2.4.3

### DISCUSSION THREAD

**Discussion by jingxuan98** [WalletConnect/walletconnect-monorepo#2848] [Developer Support] [answered]
I am using 

        "@walletconnect/sign-client": "^2.8.4",
        "@walletconnect/types": "^2.8.4",
        "@walletconnect/universal-provider": "^2.8.4",
        "@web3modal/standalone": "^2.4.3",

but getting this error when connect to Rainbow 
<img width="361" alt="Screenshot 2023-06-29 at 12 02 11 AM" src="https://github.com/WalletConnect/walletconnect-monorepo/assets/43902263/fbaa7744-de5b-4e89-ae30-f6877ac36364">

heres my way of init
```
this._walletConnectProvider = await UniversalProvider.init({
            projectId: DEFAULT_PROJECT_ID,
            logger: logLevel,
            relayUrl: relayURL,
        });

this._web3Modal = new Web3Modal({
            projectId: DEFAULT_PROJECT_ID,
            walletConnectVersion: 2,
          });

        this.session = await this._walletConnectProvider.connect({
            namespaces: {
                eip155: {
                    methods: [
                        "eth_sendTransaction",
                        "eth_signTransaction",
                        "eth_sign",
                        "personal_sign",
                        "eth_signTypedData",
                    ],
                    chains: [`eip155:${chainId}`],
                    events: ["chainChanged", "accountsChanged"],
                    rpcMap: {chainId: `https://rpc.ankr.com/eth_goerli`},
                }
            }
        });

        const accounts = await this._walletConnectProvider.enable();
        if(accounts.length) {
            this._state.accounts = accounts
            this._state.isConnected = true
            this.provider = new providers.Web3Provider(this._walletConnectProvider, 'any')    
        }
```

**Comment by xzilja** [comment]
You might want to look into:

(html)
https://docs.walletconnect.com/2.0/web/walletConnectModal/sign/html/installation

(react)
https://docs.walletconnect.com/2.0/web/walletConnectModal/sign/react/installation

They provide simpler / bundled implementation for what you are trying to acheive.

**Reply by jingxuan98** [reply]
Thanks! but our team make all of this as a class component to get the signers and handle the wallet connect part which is used for other parts in our project as well, so i found using @walletconnect/universal-provider is the easiest way for us to speedup the migration, maybe in future but guess not rn switching to the walletConnectModal 


Back to the topic, any idea how can i solve the issue above?

**Comment by glitch-txs** [comment, marked as answer]
Try removing "eth_signTransaction", there's something in the proposal that's not supported by the wallet, you can also move some to optional namespaces.

**Reply by jingxuan98** [reply]
Thanks ! this works!
